### DISCUSSION TITLE
How to return a pointer to a Client, or the equivalence of this (Doing it for me returns the interface)

### DISCUSSION THREAD

**Discussion by oAnubis** [redis/rueidis#175] [Q&A] [answered]
Seems if you try to return a pointer to the client it returns a pointer to an interface. Extra code seems to be required in this case to make this usable with DDD or Microservice architecture.  

I am sure this is not actually the case; I am just missing something. For context I am coming from Go-Redis, testing other drivers. This one looks better overall, especially that the returns are not interface slices, which are a pain to work with (In comparison to other Go types).

**Comment by rueian** [comment, marked as answer]
I would recommend just storing and passing the `Client` interface around. Just like the go-redis’ UniversalClient.

In general, there would be no need to hold a pointer value of the concrete implementation struct.

May I have some of extra code you mentioned to understand your problem in more detail?
